File: corpus/public-domain/dump/C/oldVersions/brute.c
/*
 * brute.c by Jigsy (https://github.com/Jigsy1) released under the Unlicense.
 *
 * Compiled on Windows 8.1 using TCC available from: https://bellard.org/tcc/
 *   And using Pelles C available from: http://www.smorgasbordet.com/pellesc/
 *
 * How to:
 * ----------
 * ...> tcc -c brute.c
 * ...> tcc -run brute.o
 *
 *
 * Note: 26^4 - or any number higher than four will not work - on TCC. It'll get to about eomb and just end.
 *	 However, compiling under Pelles had no problems getting to zzzz.
 */

#include <stdio.h>
#include <string.h>

void rotateChar(int stringIndex, int mapIndex, int again);

/* END OF PROTOTYPES */

int breakFlag = 0;
const char CHAR_MAP[] = "abcdefghijklmnopqrstuvwxyz";
char string[] = "aaa";

void rotateChar(int stringIndex, int mapIndex, int again) {
	if (string[stringIndex] == CHAR_MAP[(strlen(CHAR_MAP) - 1)]) {
		mapIndex = 0;
		int newIndex = (stringIndex + 1);
		string[stringIndex] = CHAR_MAP[mapIndex];
		if (newIndex < strlen(string)) {
			char *tempChar;
			int tempIndex;
			tempChar = strchr(CHAR_MAP, string[newIndex]);
			tempIndex = (int) (tempChar - CHAR_MAP);
			rotateChar(newIndex, (tempIndex + 1), 0);
		}
		else {
			breakFlag = 1;
			printf("\nEnd.\n");
		}
	}
	string[stringIndex] = CHAR_MAP[mapIndex];
	if (again > 0) {
		if (breakFlag == 0) {
			printf("%s\n", string);
			rotateChar(stringIndex, (mapIndex + 1), 1);
		}
	}
}

int main(void) {
	rotateChar(0, 0, 1);	// Start from 0, from the 1st character ("a", which is 0 in the array), and rotate again.
	return 0;
}

// EOF

File: corpus/public-domain/dump/py3/oldVersions/brute.py
# brute.py by Jigsy (https://github.com/Jigsy1) released under the Unlicense.
#
# Just a demo of looping through letter(s). aaaa->baaa->caaa->...->yzzz->zzzz


# import sys


# define

# CHAR_MAP = "abcdefghijklmnopqrstuvwxyz"
# STR_LEN = 3

CHAR_MAP = "abcd"
STR_LEN = len(CHAR_MAP)


# Core

# sys.setrecursionlimit(20000)
# ¦-> I would avoid increasing this beyond 20,000 if you use 26*26*26. (Which is 17,576 anyway.)
# `-> And 4^4 is about 256, so that should be enough for this demo.

breakFlag = []
# `-> DO NOT ADD ANYTHING TO THIS!
string = list(CHAR_MAP[0] * STR_LEN)

def rotateChar(stringIndex, mapIndex, again):
        if string[stringIndex] == CHAR_MAP[(len(CHAR_MAP) - 1)]:
                mapIndex = 0
                newIndex = (stringIndex + 1)
                string[stringIndex] = CHAR_MAP[mapIndex]
                if newIndex < len(string):
                        tempIndex = CHAR_MAP.rfind(string[newIndex])
                        rotateChar(newIndex, (tempIndex + 1), 0)
                else:
                        # ,-> As I can't find any other way to end the recursion loop as "return" doesn't seem to be
                        # ¦->  doing anything, and "break" doesn't work outside of a loop... and Python doesn't have "goto"
                        # ¦->  (Seriously!? goto is useful in some cases! (Like this one! >_<))
                        # ¦->  we'll just set a hacky flag and stop the loop that way.
                        breakFlag.append('1')
                        print("\nEnd.")
        string[stringIndex] = CHAR_MAP[mapIndex]
        if again > 0:
                if len(breakFlag) == 0:
                        print(string)
                        rotateChar(stringIndex, (mapIndex + 1), 1)

def main():
        rotateChar(0, 0, 1)
        # `-> Start at 0, from the 1st character ('a') which is 0 in arrayspeak, and rotate again.

if __name__ == "__main__":
	main()


# EOF
